File: components/albums/src/main/java/org/superbiz/moviefun/CsvUtils.java
package org.superbiz.moviefun;

import com.fasterxml.jackson.databind.MappingIterator;
import com.fasterxml.jackson.databind.ObjectReader;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

public class CsvUtils {

    public static <T> List<T> readFromCsv(ObjectReader objectReader, InputStream inputStream) throws IOException {
        List<T> results = new ArrayList<>();

        MappingIterator<T> iterator = objectReader.readValues(inputStream);

        while (iterator.hasNext()) {
            results.add(iterator.nextValue());
        }

        return results;
    }
}

File: components/albums/src/main/java/org/superbiz/moviefun/CsvUtils.kt
package org.superbiz.moviefun

import com.fasterxml.jackson.databind.ObjectReader
import java.io.IOException
import java.io.InputStream
import java.util.*

object CsvUtils {
    @Throws(IOException::class)
    fun <T> readFromCsv(objectReader: ObjectReader, inputStream: InputStream) = ArrayList<T>().apply {
        val iterator = objectReader.readValues<T>(inputStream)

        while (iterator.hasNext()) {
            add(iterator.nextValue())
        }
    }
}
